docs/proposals/versioning-and-release.md

```diff
@@ -0,0 +1,72 @@
+# Devfile Versioning and Release Process
+This design document outlines the proposed versioning and release process for the Devfile spec. 
+
+This document summarizes parts from he Devfile API technical meeting slides presented by @davidfestal back in October, and I recommend having a read through of it: https://docs.google.com/presentation/d/1ohM1HzPB59a3ajvB7rVWJkKONLBAuT0mb5P20_A6vLs/edit#slide=id.g8fc722bef9_1_8
+
+## Versioning
+
+The following sections outline how we version the Devfile Kubernetes API, as well as the JSON schema generated from the API.
+
+### Kubernetes API
+The Devfile Kubernetes API (defined in https://github.com/devfile/api/) is the single source of truth for the Devfile spec. It’s defined under `pkg/apis/` and contains the Go structs that are used by the devworkspace operator and devfile library. It's also where the Devfile JSON schema is generated from. 
+
+**Versioning Scheme**: [Kubernetes](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning) (e.g. v1alpha1, v1beta2, v1, v2, etc)
+
+**Update Process**: 
```

Comment:
Consider changing the order of "Update Process" and "When to Update". I feel more natural to read in that order.

Reply:
Sounds good to me, updated 👍 .